src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionService.java

```diff
@@ -44,20 +44,41 @@ void extract(Table table, BigQueryExtractionData extractionData) {
   }
 
   void cleanup(BigQueryExtractionData extractionData) {
-    String dataBucket = extractionData.getDataBucket();
-    String dataKey = extractionData.getDataKey();
-    String format = extractionData.getFormat();
-    String dataUri = extractionData.getDataUri();
+    deleteBucketAndContents(extractionData.getDataBucket());
+  }
+
+  private void deleteBucketAndContents(String dataBucket) {
+    deleteObjectsInBucket(dataBucket);
+    deleteBucket(dataBucket);
+  }
 
-    BlobId blobId = BlobId.of(dataBucket, dataKey + "." + format);
-    boolean suceeded = storage.delete(blobId);
-    if (!suceeded) {
-      log.warn("Could not delete object {}", dataUri);
+  private void deleteBucket(String dataBucket) {
+    try {
+      Bucket bucket = storage.get(dataBucket);
+      boolean suceeded = bucket.delete();
+      if (suceeded) {
+        log.info("Deleted bucket {}", dataBucket);
+      } else {
+        log.warn("Could not delete bucket {}", dataBucket);
+      }
+    } catch (Exception e) {
+      log.warn(e.toString());
     }
-    Bucket bucket = storage.get(dataBucket);
-    suceeded = bucket.delete();
-    if (!suceeded) {
-      log.warn("Could not delete bucket {}", dataBucket);
+  }
+
+  private void deleteObjectsInBucket(String dataBucket) {
+    try {
+      Iterable<Blob> blobs = storage.list(dataBucket).iterateAll();
+      for (Blob blob : blobs) {
+        boolean suceeded = storage.delete(blob.getBlobId());
+        if (suceeded) {
+          log.info("Deleted object {}", blob);
+        } else {
+          log.warn("Could not delete object {}", blob);
+        }
+      }
+    } catch (Exception e) {
+      log.warn(e.toString());
```

Comment:
Shouldn't we log the exception too? i.e. `log.warn("Error deleting objects in bucket {}", bucket, e)`.